src/test/java/de/tum/in/ase/parser/IntegrationTest.java

```diff
@@ -16,69 +21,97 @@
-
-    private static final String EXPECTED_INVALID_FILENAME = "{\"tool\":null,\"issues\":[{\"filePath\":\"cpd_invalid.txt\",\"startLine\":1,\"rule\":\"ExceptionDuringParsing\",\"category\":\"miscellaneous\",\"message\":\"An exception occurred during parsing the report for file cpd_invalid.txt. Exception: java.lang.IllegalArgumentException: File must be xml format\"}]}";
-
-    private static final String EXPECTED_INVALID_XML = "{\"tool\":null,\"issues\":[{\"filePath\":\"invalid_xml.xml\",\"startLine\":1,\"rule\":\"ExceptionDuringParsing\",\"category\":\"miscellaneous\",\"message\":\"An exception occurred during parsing the report for file invalid_xml.xml. Exception: %s\"}]}";
-
-    private static final String EXPECTED_INVALID_NAME = "{\"tool\":null,\"issues\":[{\"filePath\":\"invalid_name.xml\",\"startLine\":1,\"rule\":\"ExceptionDuringParsing\",\"category\":\"miscellaneous\",\"message\":\"An exception occurred during parsing the report for file invalid_name.xml. Exception: de.tum.in.ase.parser.exception.UnsupportedToolException: Tool for identifying tag data not found\"}]}";
+        ReportParser parser = new ReportParser();
+        String actual = parser.transformToJSONReport(toolReport);
+
+        try {
+            BufferedReader reader = Files.newBufferedReader(Paths.get(EXPECTED_FOLDER_PATH + expectedJSONReportFileName));
+            String expected = reader.readLine();
+            assertEquals(expected, actual);
```

Comment:
assertions are cough by the catch:
https://rules.sonarsource.com/java/RSPEC-5779

Perhaps we should create a new method that reads in the data and then it gets compared here.

Reply:
Alternatively, don’t use any try-catch for failing to read the file neither here nor in the `@Test` methods and instead forward the exception. When the `@Test` throws the `IOException` the test is marked as failed, which is exactly what we want here.

Reply:
I'd prefer throwing the exception actually as well. We already throw the `ParserException` and the behavior fits well.